pycheribuild/projects/cross/newlib.py

```diff
@@ -118,31 +120,40 @@ def configure(self):
             newlib_cv_ldbl_eq_dbl="yes",
             LD_FOR_TARGET=str(self.target_info.linker), LDFLAGS_FOR_TARGET="-fuse-ld=" + str(self.target_info.linker),
             )
-        self.configureArgs.extend([
-            "--enable-malloc-debugging",
-            "--enable-newlib-long-time_t",  # we want time_t to be long and not int!
-            "--enable-newlib-io-c99-formats",
-            "--enable-newlib-io-long-long",
-            # --enable-newlib-io-pos-args (probably not needed)
-            "--disable-newlib-io-long-double",  # we don't need this, MIPS long double == double
-            "--enable-newlib-io-float",
-            # "--disable-newlib-supplied-syscalls"
-            "--disable-libstdcxx",  # not sure if this is needed
-
-            # we don't have any multithreading support on baremetal
-            "--disable-newlib-multithread",
-
-            "--enable-newlib-global-atexit",  # TODO: is this needed?
-            # --enable-newlib-nano-malloc (should we do this?)
-            "--disable-multilib",
-
-            # TODO: smaller lib? "--enable-target-optspace"
-
-            # FIXME: these don't seem to work
-            "--enable-serial-build-configure",
-            "--enable-serial-target-configure",
-            "--enable-serial-host-configure",
-        ])
+
+        if self.target_info.is_baremetal:
+          self.configureArgs.extend([
+              "--enable-malloc-debugging",
+              "--enable-newlib-long-time_t",  # we want time_t to be long and not int!
+              "--enable-newlib-io-c99-formats",
+              "--enable-newlib-io-long-long",
+              # --enable-newlib-io-pos-args (probably not needed)
+              "--disable-newlib-io-long-double",  # we don't need this, MIPS long double == double
+              "--enable-newlib-io-float",
+              # "--disable-newlib-supplied-syscalls"
+              "--disable-libstdcxx",  # not sure if this is needed
+
+              # we don't have any multithreading support on baremetal
+              "--disable-newlib-multithread",
+
+              "--enable-newlib-global-atexit",  # TODO: is this needed?
+              # --enable-newlib-nano-malloc (should we do this?)
+              "--disable-multilib",
+
+              # TODO: smaller lib? "--enable-target-optspace"
+
+              # FIXME: these don't seem to work
+              "--enable-serial-build-configure",
+              "--enable-serial-target-configure",
+              "--enable-serial-host-configure",
+          ])
+        elif self.target_info.is_rtems:
+          self.configureArgs.extend([
+              "--enable-newlib-io-c99-formats",
+              "--disable-libgloss",
+              "--disable-static", "--enable-shared",
```

Comment:
I'm a bit surprised by the --enable-shared? Don't we wan't it static?

Reply:
Nice catch. Yeah, definitely should be static. I fail to remember where/why did these flags come from, but they don't seem to be doing what they should (i.e., I always get static libs). I deleted them.